darwin/future/meta/queries/workflow.py

```diff
@@ -43,38 +44,38 @@ def collect(self) -> List[Workflow]:
 
         return workflows
 
-    def _execute_filters(self, workflows: List[Workflow], filter: QueryFilter) -> List[Workflow]:
+    def _execute_filters(self, workflows: List[WorkflowMeta], filter: QueryFilter) -> List[WorkflowMeta]:
         if filter.name == "id":
             id_to_find = UUID(filter.param)
             return [w for w in workflows if w.id == id_to_find]
 
         if filter.name == "inserted_at_start":
             start_date = datetime.fromisoformat(filter.param)
-            return [w for w in workflows if self._date_compare(w.inserted_at, start_date)]
+            return [w for w in workflows if w._item is not None and self._date_compare(w._item.inserted_at, start_date)]
```

Comment:
Maybe the `is not None` could've been reasonably added to the `_date_compare` function, but equally I'm not too bothered

Reply:
This is actually mostly for mypy here. It complains because _item is Optional[T] and thus it'll squiggly line everything without some check in the list comprehension.

Reply:
but we could change the typing on _date_compare to take an Optional and check it in there as well